ios/brave-ios/Tests/GrowthTests/DAUTests.swift

```diff
@@ -266,9 +266,10 @@ class DAUTests: XCTestCase {
     let yearString = Date().mondayOfCurrentWeekFormatted?.truncate(length: 4, trailing: "")
     let year = Int(yearString!)!
 
-    // There is no easy way to mock a different calendar in unit tests, just checking if
-    // year we get looks 'correct' should be good enough.
-    XCTAssert(year > 2018 && year < 2025)
+    // Verify the year in `mondayOfCurrentWeekFormatted` matches current gregorian calendar year.
+    let gregorianCalendar = Calendar(identifier: .gregorian)
+    let gregorianCalendarYear = gregorianCalendar.dateComponents([.year], from: Date()).year
+    XCTAssertEqual(gregorianCalendarYear, year)
```

Comment:
Wouldn't this fail if the monday of the current week is in a previous year? For example from 2024-2025, if the current date was Jan 1 2025 (Wed), the monday of the week would be Dec 30 2024

Reply:
Hmm yeah I think you're right since we're looking backwards. Maybe better to check against `gregorianCalendarYear - 1` for that case, ex:
```XCTAssert(year >= (gregorianCalendarYear - 1) && year <= gregorianCalendarYear)```

Reply:
Either that or use a fixed date for generating the current monday (which honestly may be safer anyways)